### PR TITLE
Move metrics server to seed

### LINKED ISSUE #2125: Move metrics-server into cluster namespace
**User Story**
Move the metrics-server into the cluster namespace. Right now we deploy it into the user cluster - which has a few downsides:
- Customer is responsible for the pod to be available - otherwise metrics are not available.
This is potentially bad during node updates as the pod might be rescheduled & relevant metrics for rescheduling are not available.
- The metrics server can be considered as integral component of Kubernetes as things like Autoscaling do not work without it. It should be a managed component - like the Kubernetes APIserver - and not just be a running pod inside the user cluster.

**Acceptance criteria**
- [ ] metrics server runs inside the cluster namespace
- [ ] horizontal pod autoscaling works

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
This PR moves the metrics server into the seed cluster.

Fixes #2125

**Release note**:
```release-note monitoring
[ACTION REQUIRED] move the metrics-server into the seed cluster. The metrics-server addon must be removed from the list of addons to install.
```

### CODE REVIEW

File: api/pkg/controller/cluster/userclusterresources.go [hunk 1]
```diff
@@ -125,13 +128,14 @@ func (cc *Controller) userClusterEnsureRoleBindings(c *kubermaticv1.Cluster, cli
...
 		machinecontroller.KubePublicRoleBinding,
+		metricsserver.RolebindingAuthReader,
 	}
 
 	for _, create := range creators {
 		var existing *rbacv1.RoleBinding
 		rb, err := create(nil, nil)
 		if err != nil {
-			return fmt.Errorf("failed to build RoleBinding: %v", err)
+			return fmt.Errorf("failed to build RolebindingAuthReader: %v", err)
```
Comment (alvaroaleman) [inline api/pkg/controller/cluster/userclusterresources.go]: Leftover from debugging?

File: api/pkg/controller/cluster/userclusterresources.go [hunk 2]
```diff
@@ -140,25 +144,25 @@ func (cc *Controller) userClusterEnsureRoleBindings(c *kubermaticv1.Cluster, cli
 			}
 
 			if _, err = client.RbacV1().RoleBindings(rb.Namespace).Create(rb); err != nil {
-				return fmt.Errorf("failed to create RoleBinding %s in namespace %s: %v", rb.Name, rb.Namespace, err)
+				return fmt.Errorf("failed to create RolebindingAuthReader %s in namespace %s: %v", rb.Name, rb.Namespace, err)
```
Comment (alvaroaleman) [inline api/pkg/controller/cluster/userclusterresources.go]: Leftover from debugging?

Comment (alvaroaleman) [reply api/pkg/controller/cluster/userclusterresources.go]: Just imagine this comment for every occurence of `RolebindingAuthReader` in an error or glog message

File: api/metrics-server/in-cluster-apiservice.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+apiVersion: apiregistration.k8s.io/v1beta1
```
Comment (alvaroaleman) [inline api/metrics-server/in-cluster-apiservice.yaml]: Where do these files in `api/metrics-server` come from?

Comment (mrIncompetent) [reply api/metrics-server/in-cluster-apiservice.yaml]: Leftovers

Comment (mrIncompetent) [reply api/metrics-server/in-cluster-apiservice.yaml]: fixed with https://github.com/kubermatic/kubermatic/pull/2320/commits/6887ee946a6de723890267114aab0e684747f81f

File: api/pkg/controller/cluster/userclusterresources.go (hunk 2 above)
Comment (mrIncompetent) [reply api/pkg/controller/cluster/userclusterresources.go]: fixed with https://github.com/kubermatic/kubermatic/pull/2320/commits/751ceb676d887096980799819a3c963cda1b450f

File: api/pkg/controller/cluster/userclusterresources.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/controller/cluster/userclusterresources.go]: fixed with https://github.com/kubermatic/kubermatic/pull/2320/commits/751ceb676d887096980799819a3c963cda1b450f

**Comment by mrIncompetent** [comment]
@alvaroaleman PTAL

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**